Wikipedia/Code/NotificationsCenterViewController.swift

```diff
@@ -10,37 +10,49 @@ final class NotificationsCenterViewController: ViewController {
     }
 
     let viewModel: NotificationsCenterViewModel
+    
+    typealias DataSource = UICollectionViewDiffableDataSource<NotificationsCenterSection, NotificationsCenterCellViewModel>
+    typealias Snapshot = NSDiffableDataSourceSnapshot<NotificationsCenterSection, NotificationsCenterCellViewModel>
+    private var dataSource: DataSource?
+    private let snapshotUpdateQueue = DispatchQueue(label: "org.wikipedia.notificationcenter.snapshotUpdateQueue", qos: .userInteractive)
+    
+    private let editTitle = WMFLocalizedString("notifications-center-edit-button-edit", value: "Edit", comment: "Title for navigation bar button to turn on edit mode for toggling notification read status")
+    private let doneTitle = WMFLocalizedString("notifications-center-edit-button-done", value: "Done", comment: "Title for navigation bar button to turn off edit mode for toggling notification read status")
+    private lazy var editButton = {
+        return UIBarButtonItem(title: editTitle, style: .plain, target: self, action: #selector(userDidTapEditButton))
+    }()
 
     // MARK: - Lifecycle
 
     @objc
     init(theme: Theme, viewModel: NotificationsCenterViewModel) {
         self.viewModel = viewModel
         super.init(theme: theme)
+        viewModel.delegate = self
     }
 
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
-	override func loadView() {
-		view = NotificationsCenterView(frame: UIScreen.main.bounds)
-		scrollView = notificationsView.collectionView
-	}
+    override func loadView() {
+        view = NotificationsCenterView(frame: UIScreen.main.bounds)
+        scrollView = notificationsView.collectionView
+    }
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
         notificationsView.apply(theme: theme)
 
-		title = CommonStrings.notificationsCenterTitle
-		setupBarButtons()
-
-		notificationsView.collectionView.delegate = self
-		notificationsView.collectionView.dataSource = self
-
-		viewModel.fetchNotifications(collectionView: notificationsView.collectionView)
-	}
+        title = CommonStrings.notificationsCenterTitle
+        setupBarButtons()
+        
+        setupCollectionView()
+        setupDataSource()
+        configureEmptyState(isEmpty: true)
```

Comment:
@staykids Thanks for pointing that out. I think we need to show this importing state in a delayed manner, like after a delay of 1 second if it's still importing, display it. But if data populates quickly don't bother to avoid a flash. I started to explore debouncing with `performSelector` and `NSObject.cancelPreviousPerformRequests` like we do elsewhere in the app but it started to feel really gross. If you don't mind I'd rather tackle it in a later PR (separately or whenever we revisit empty states for the filter status).  For now I commented it out and left a TODO note.